### PR TITLE
[Organizations] Auditing for organizations

### PR DESCRIPTION
**Pull request by scottbommarito**
https://github.com/NuGet/NuGetGallery/issues/5113

### CODE REVIEW

File: src/NuGetGallery.Core/Auditing/UserAuditRecord.cs [hunk 1]
```diff
@@ -17,23 +17,15 @@ public class UserAuditRecord : AuditRecord<AuditedUserAction>
         public CredentialAuditRecord[] Credentials { get; }
         public CredentialAuditRecord[] AffectedCredential { get; }
         public string AffectedEmailAddress { get; }
+        public string AffectedMemberUsername { get; }
+        public bool AffectedMemberIsAdmin { get; }
```
Comment (chenriksson) [inline src/NuGetGallery.Core/Auditing/UserAuditRecord.cs]: Maybe add comments above properties that are specific to only org (membership) or user (credential) accounts.

File: src/NuGetGallery.Core/Auditing/UserAuditRecord.cs [hunk 2]
```diff
@@ -45,25 +37,40 @@ public UserAuditRecord(User user, AuditedUserAction action, IEnumerable<Credenti
...
+        public UserAuditRecord(User user, AuditedUserAction action, Credential affected)
+            : this(user, action, new[] { affected })
+        {
+        }
+
+        public UserAuditRecord(User user, AuditedUserAction action, IEnumerable<Credential> affected)
+            : this(user, action)
+        {
+            AffectedCredential = affected.Select(c => new CredentialAuditRecord(c, action == AuditedUserAction.RemoveCredential)).ToArray();
         }
```
Comment (chenriksson) [inline src/NuGetGallery.Core/Auditing/UserAuditRecord.cs]: Why the refactoring?

Comment (scottbommarito) [reply src/NuGetGallery.Core/Auditing/UserAuditRecord.cs]: It was really awkward before...most of the constructors were passing `Enumerable.Empty()` when they called `this(...)`. This is the proper way for these constructors to be set up.

File: src/NuGetGallery.Core/Auditing/UserAuditRecord.cs (hunk 1 above)
Comment (scottbommarito) [reply src/NuGetGallery.Core/Auditing/UserAuditRecord.cs]: Sure. Will also add to the other fields that are only used for some calls.

**Comment by scottbommarito** [comment]
@chenriksson Shims PR coming soon!

**Pull request merged**